dandiapi/urls.py

```diff
@@ -68,6 +69,11 @@ def to_url(self, value):
 register_converter(DandisetIDConverter, 'dandiset_id')
 urlpatterns = [
     path('api/', include(router.urls)),
+    re_path(
+        r'api/assets/(?P<asset_id>[0-9a-f\-]{36})/',
```

Comment:
looks good to me. i'm assuming `assets/<id>` and `assets/<id>/` both will work here. and django will do the magic of adding the `/` if it's missing.

Reply:
IIRC that is not the case... yeap -- magic doesn't exist:
```
$> curl https://api.dandiarchive.org/api/dandisets/000027/versions/draft/assets/1c095f5f-d1e2-45db-b807-fdcfea08c6de/           
{"id":"dandiasset:1c095f5f-d1e2-45db-b807-fdcfea08c6de","path":"sub-RAT123/sub-RAT123.nwb","access":[{"status":"dandi:OpenAccess","schemaKey":"AccessRequirements"}],"digest":{"dandi:sha2-256":"1a765509384ea96b7b12136353d9c5b94f23d764ad0431e049197f7875eb352c","dandi:dandi-etag":"6ec084ca9d3be17ec194a8f700d65344-1"},"@context":"https://raw.githubusercontent.com/dandi/schema/master/releases/0.4.4/context.json","approach":[],"schemaKey":"Asset","contentUrl":["https://api.dandiarchive.org/api/assets/1c095f5f-d1e2-45db-b807-fdcfea08c6de/download/","https://dandiarchive.s3.amazonaws.com/blobs/2db/af0/2dbaf0fd-5003-4a0a-b4c0-bc8cdbdb3826"],"identifier":"1c095f5f-d1e2-45db-b807-fdcfea08c6de","repository":"https://dandiarchive.org/","contentSize":18792,"dateModified":"2021-06-29T18:59:29.491552-04:00","schemaVersion":"0.4.4","encodingFormat":"application/x-nwb","wasGeneratedBy":[{"id":"urn:uuid:e664a279-71dc-4290-9bb1-1c843e2b19cf","name":"Metadata generation","schemaKey":"Activity","description":"Metadata generated by DANDI cli","wasAssociatedWith":[{"url":"https://github.com/dandi/dandi-cli","name":"DANDI Command Line Interface","version":"0.21.0","schemaKey":"Software","identifier":"RRID:SCR_019009"}]}],"wasAttributedTo":[{"age":{"value":"P12M","unitText":"Years from birth","schemaKey":"PropertyValue"},"sex":{"name":"Male","schemaKey":"SexType","identifier":"http://purl.obolibrary.org/obo/PATO_0000384"},"species":{"name":"Brown rat","schemaKey":"SpeciesType","identifier":"http://purl.obolibrary.org/obo/NCBITaxon_10116"},"genotype":"WT","schemaKey":"Participant","identifier":"RAT123"}],"blobDateModified":"2020-10-21T10:10:35.457789-04:00","variableMeasured":[],"measurementTechnique":[]}%                                                                                            1 9731.....................................:Thu 29 Jul 2021 04:40:55 PM EDT:.
lena:~
$> curl https://api.dandiarchive.org/api/dandisets/000027/versions/draft/assets/1c095f5f-d1e2-45db-b807-fdcfea08c6de
1 9732.....................................:Thu 29 Jul 2021 04:40:56 PM EDT:.
lena:~
$> echo $?
0
```

Reply:
the server uses a redirect mechanism:

> curl -L https://api.dandiarchive.org/api/dandisets/000027/versions/draft/assets/1c095f5f-d1e2-45db-b807-fdcfea08c6de

will work.

Reply:
I knew I should have just used wget ;) thanks @satra